pkg/node/nodes_test.go

```diff
@@ -0,0 +1,84 @@
+package node
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+var _ = Describe("MaxUnavailable nodes", func() {
+	type maxUnavailableCase struct {
+		nmstateEnabledNodes          int
+		maxUnavailable               intstr.IntOrString
+		expectedScaledMaxUnabailable int
+		expectedError                bool
+	}
+	DescribeTable("testing ScaledMaxUnavailableNodeCount",
+		func(c maxUnavailableCase) {
+			maxUnavailable, err := ScaledMaxUnavailableNodeCount(c.nmstateEnabledNodes, c.maxUnavailable)
+			if c.expectedError {
+				Expect(err).Should(HaveOccurred())
+			} else {
+				Expect(err).ShouldNot(HaveOccurred())
+			}
```

Comment:
Here we just exercise the GomegaMatcher
```suggestion
                        Expect(err).Should(c.expectedError)
```

Reply:
Done